src/tests/ftest/cart/selftest/cart_selftest_three_node.py

```diff
@@ -61,5 +61,10 @@ def test_cart_selftest(self):
                 self, self.env, "test_clients", index=index)
             self.utils.launch_test(self, clicmd, srv_rtn)
 
+        # Give few seconds for servers to fully shut down before exiting
+        # from this test.
+        if not self.utils.wait_process(srv_rtn, 5):
```

Comment:
what if you did this in a loop instead with shorter sleeps and kill after sufficient iterations?

Reply:
wait_process internally is already a loop that sleeps for 1 sec and polls until timeout sec specified